src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,129 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private int capacity = 10;
+    private int size;
+
+    private T[] elementData;
+
+    public ArrayList() {
+        this.elementData = (T[]) new Object[capacity];
+    }
+
+    public ArrayList(int arrayListCapacity) {
+        this.elementData = (T[]) new Object[arrayListCapacity];
+        capacity = arrayListCapacity;
+    }
+
+    private void increaseCapacity() {
+        if (this.size == elementData.length) {
```

Comment:
```suggestion
        if (size == elementData.length) {
```
`this` keyword is redundant when we have no shadowing (2 variables with same name but different scopes)